Update certs for cases where support for DES cipher is not available

rado17 opened the pull request:
Update certs for cases where support for DES cipher is not available.

rado17 committed:
snippets: wifi-enterprise: Increase MBEDTLS heap size

With NRF security, higher heap requirement is seen for operating
with RSA-3072 based certificates. Add a NRF SoC specific conf in
snippets to handle this.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

rado17 committed:
samples: net: wifi: Add AES-only certificates

Add AES-only cipher certificates. Useful for systems which
have only AES support and DES is not enabled.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

krish2718 commented on snippets/wifi-enterprise/wifi-enterprise_nrf.conf:
```diff
@@ -0,0 +1,17 @@
+# Enable Wi-Fi enterprise mode
```
you don't need to duplicate entire file, see https://docs.zephyrproject.org/latest/build/snippets/writing.html#board-specific-settings we can only add overrides here

rado17 replied:
Corrected.

krish2718 commented on snippets/wifi-enterprise/snippet.yml:
```diff
@@ -1,3 +1,8 @@
+boards:
+  /.*/nrf.*/cpuapp/:
+    append:
+      EXTRA_CONF_FILE: wifi-enterprise_nrf.conf
```
`overlay-nrf-wifi.conf`?

rado17 committed:
snippets: wifi-enterprise: Store certificates in Protected storage

Add config options required for storing certificates in protected
storage, and to free up space of ROM to accommodate the additional
features that are enabled for TFM.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

rado17 commented on snippets/wifi-enterprise/wifi-enterprise-nrf-ns.conf:
```diff
@@ -0,0 +1,8 @@
+# For TLS and X.509 processing MbedTLS needs large heap size and using separate heap
+# for MbedTLS gives us more control over the heap size.
+CONFIG_MBEDTLS_HEAP_SIZE=75000
```
Verified that it is needed.

rado17 replied:
To clarify, CONFIG_MBEDTLS_ENABLE_HEAP is needed and enabled in default snippet conf (https://github.com/zephyrproject-rtos/zephyr/blob/366d45fd34fe536747d4883c4673d83249386ac0/snippets/wifi-enterprise/wifi-enterprise.conf#L7)

MaochenWang1 commented:
why need to add these rsa2k_no_des PEM files, since no code change to directly use them?
Currently runtime cert is supported, or you can replace the certs and build image for DES cipher not available case.
I am not sure it's a good idea to add several kinds of cert/key files here

krish2718 commented:
> why need to add these rsa2k_no_des PEM files, since no code change to directly use them? Currently runtime cert is supported, or you can replace the certs and build image for DES cipher not available case. I am not sure it's a good idea to add several kinds of cert/key files here

Even for build time we now have an option to override, but the intention is to have a golden certs that work out of box, I guess these 3 variants now should cover all platforms.
